apollo-adminservice/src/main/scripts/startup.sh

```diff
@@ -53,6 +56,35 @@ if [ "$windows" == "1" ] && [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java"
     echo "Windows new JAVA_HOME is: $JAVA_HOME"
 fi
 
+# Find Java
+if [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java" ]]; then
+    javaexe="$JAVA_HOME/bin/java"
+elif type -p java > /dev/null 2>&1; then
+    javaexe=$(type -p java)
+elif [[ -x "/usr/bin/java" ]];  then
+    javaexe="/usr/bin/java"
+else
+    echo "Unable to find Java"
+    exit 1
+fi
+
+if [[ "$javaexe" ]]; then
+    version=$("$javaexe" -version 2>&1 | awk -F '"' '/version/ {print $2}')
+    version=$(echo "$version" | awk -F. '{printf("%03d%03d",$1,$2);}')
+    # now version is of format 009003 (9.3.x)
+    if [ $version -ge 011000 ]; then
+        JAVA_OPTS="$JAVA_OPTS -Xlog:gc*:$LOG_DIR/gc.log -Xlog:safepoint -Xlog:gc+heap=trace"
+    elif [ $version -ge 010000 ]; then
+        JAVA_OPTS="$JAVA_OPTS -Xlog:gc*:$LOG_DIR/gc.log -Xlog:safepoint -Xlog:gc+heap=trace"
+    elif [ $version -ge 009000 ]; then
+        JAVA_OPTS="$JAVA_OPTS -Xlog:gc*:$LOG_DIR/gc.log -Xlog:safepoint -Xlog:gc+heap=trace"
```

Comment:
It's better to print out the `Current time and date`([2019-04-06T20:48:47.390-0800]) instead of `Time since the start of the JVM in seconds and milliseconds.`([0.444s]). So it's better to decorate the gc output like `-Xlog:gc*:$LOG_DIR/gc.log:time,level,tags`. See [Decorations](https://docs.oracle.com/javase/9/tools/java.htm#JSWOR-GUID-BE93ABDC-999C-4CB5-A88B-1994AAAC74D5#GUID-BE93ABDC-999C-4CB5-A88B-1994AAAC74D5__DECORATIONS-A7A4AD31)